usecase/github_user.go

```diff
@@ -0,0 +1,25 @@
+//Package usecase processes the information from repository, making any needed calculations, data transformation, etc.
+//If it recieved data it's from controllers, coming from delivery layer
+package usecase
+
+import (
+	"lolidelgado/github-users/models"
+	"lolidelgado/github-users/repository"
+)
+
+type GithubUserUseCase struct {
+	githubUserRepo *repository.GithubUser
+}
+
+func NewGithubUser(githubUserRepo *repository.GithubUser) *GithubUserUseCase {
+	return &GithubUserUseCase{
+		githubUserRepo,
+	}
+}
+
+func (g *GithubUserUseCase) FetchAll() ([]models.GithubUser, error) {
+	return g.githubUserRepo.FetchAll()
+}
```

Comment:
Something important in Go at the time of determining the root cause of an error is to wrap errros.

Let's say you have more logic here, and you simply decide to return what you get from the repository, and also let's imagine you need to determine the reason of this error. You won't know at glance if the repo package, or the usecase package, or any other package was the pace  where the error was originated.

By wrapping the error in each layer you are adding useful information that will help debugging tasks.

To wrap errors, there's a method named `Errorf` in the `fmt`pakage.

```suggestion
func (g *GithubUserUseCase) FetchAll() ([]models.GithubUser, error) {
	githubUsers, err := g.githubUserRepo.FetchAll()
	if err != nil {
		return nil, fmt.Errorf("Error fetching github users: %w", err)
	}

	return githubUsers, nil
}

```

Reply:
Is it better if I handle the errors from the bottom, indicating errors and reasons from the repository, I am going for that approach but please let me know if there is another way that is more suggested as best practice.